Makefile

```diff
@@ -40,14 +43,19 @@ xtest:
 ta:
 	$(q)$(MAKE) -C ta CROSS_COMPILE="$(CROSS_COMPILE_TA)" \
 			  q=$(q) \
-			  O=$(out-dir)/ta \
+			  O=$(out-dir) \
 			  $@
 
 .PHONY: clean
 ifneq ($(wildcard $(TA_DEV_KIT_DIR)/host_include/conf.mk),)
 clean:
 	$(q)$(MAKE) -C host/xtest O=$(out-dir)/xtest q=$(q) $@
-	$(q)$(MAKE) -C ta O=$(out-dir)/ta q=$(q) $@
+ifneq ($O,)
+	$(q)bash -c 'cd host/xtest && [ ! -d $(O) ] || \
+			[ `stat --format=%i $(O)` == `stat --format=%i .` ] || \
+			rmdir --ignore-fail-on-non-empty  $(O)'
```

Comment:
ok for `rmdir $(O) 2> /dev/null`, but we must keep the ugly `cd host/xtest` as $(O) might be a relative part. agree ? any nicer proposal (could check if $(O) is relative, or hide complexity in a macro, but i don't feel it would be useful).